File: 10/C++/Solution.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution
{
  public:
	int jumpFloor(int number)
	{
		int plans[number];
		plans[1] = 1;
		plans[2] = 2;
		for (int cur = 3; cur <= number; cur++)
		{
			plans[cur] = plans[cur - 1] + plans[cur - 2];
		}
		return plans[number];
	}
};

int main()
{
	Solution solution;
	cout << solution.jumpFloor(3) << endl;

	return 0;
}

File: 10/Java/Solution.java
public class Solution {

    public int JumpFloor(int target) {

        if (target == -1)
            return 0;
        else if (target == 1)
            return 1;
        else if (target == 2)
            return 2;
        else
            return JumpFloor(target - 1) + JumpFloor(target - 2);

    }

    public static void main(String[] args) {

        System.out.println(new Solution().JumpFloor(3));
    }
}
